File: bindings/java/examples/Prompt18SecureMutation.java
package io.wellfriendpdf.examples;

import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import io.wellfriendpdf.Wellfriend;

public final class Prompt18SecureMutation {
    public static void main(String[] args) throws Exception {
        if (args.length != 1) {
            throw new IllegalArgumentException("usage: Prompt18SecureMutation input.pdf");
        }
        try (WellfriendPdf.Document document = WellfriendPdf.Document.open(Path.of(args[0]))) {
            System.out.println(document.prompt18ReportJson());
            System.out.println(document.associatedFilesReportJson());
            System.out.println(document.editPolicyReportJson("attachment_remove"));
            String options = "{\"filename\":\"evidence.txt\",\"mime\":\"text/plain\",\"relationship\":\"data\",\"deterministic\":true}";
            WellfriendPdf.BinaryResult added = document.associatedFileAdd(
                "bounded evidence".getBytes(StandardCharsets.UTF_8), options);
            Files.write(Path.of("prompt18-associated.pdf"), added.bytes());
        }
    }
}

File: bindings/dotnet/examples/Prompt18SecureMutation.cs
using Oxide.Sdk;

if (args.Length != 1)
{
    Console.Error.WriteLine("usage: Prompt18SecureMutation input.pdf");
    return 2;
}

using var document = OxideDocument.Open(File.ReadAllBytes(args[0]));
Console.WriteLine(document.Prompt18ReportJson());
Console.WriteLine(document.AssociatedFilesReportJson());
Console.WriteLine(document.EditPolicyReportJson("attachment_remove"));

var addOptions = """{"filename":"evidence.txt","mime":"text/plain","relationship":"data","deterministic":true}""";
var added = document.AssociatedFileAdd("bounded evidence"u8.ToArray(), addOptions);
File.WriteAllBytes("prompt18-associated.pdf", added.Bytes);
return 0;
